### ISSUE TITLE
`GM.xmlHttpRequest.onerror` causes uncaught exception when logging error

### ISSUE THREAD

**Issue by adamlui** [Tampermonkey/tampermonkey#2158]
(Please fill out the issue template with your details)

## Expected Behavior

Log the error, do not throw

## Actual Behavior

`Uncaught TypeError: can't convert label to string`

## Specifications

- Firefox: 129.0.1 (64-bit)
- TM: v5.1.1
- OS: Win10 Home

## Script

(Please give an example of the script if applicable.)

```js
...
function consoleErr(label, msg) { console.error(`${config.appSymbol} ${config.appName} » ${label}${ msg ? `: ${msg}` : '' }`)}

GM.xmlHttpRequest({
    method: apis[get.reply.api].method,
    url: apis[get.reply.api].endpoints?.completions || apis[get.reply.api].endpoint,
    responseType: config.streamingDisabled || !config.proxyAPIenabled ? 'text' : 'stream',
    headers: api.createHeaders(get.reply.api), data: api.createPayload(get.reply.api, msgChain),
    onload: resp => dataProcess.text(get.reply, resp),
    onloadstart: resp => dataProcess.stream(get.reply, resp),
    onerror: err => { consoleErr(err)
        if (!config.proxyAPIenabled) appAlert(!config.openAIkey ? 'login' : ['openAInotWorking', 'suggestProxy'])
        else if (get.reply.status != 'done') api.tryNew(get.reply)
    }
})
...
```

The above on error will throw unless `err` is wrapped in `JSON.stringify()` 

![image](https://github.com/user-attachments/assets/35354402-879e-4a3b-bc23-614b79cf4e5b)

(In Violentmonkey it does not throw + properly logs the `err` obj and script healthily keeps on running)

![image](https://github.com/user-attachments/assets/eef9d177-9ff7-41c2-acdf-e192ba5f11bf)

Maybe related to https://github.com/Tampermonkey/tampermonkey/issues/1877

**Comment by adamlui** [comment]
I fixed the throwing by updating `consoleErr()` to:

```js
function consoleErr(label, msg) {
    console.error( `${config.appSymbol} ${config.appName} » ${
        typeof label == 'object' ? JSON.stringify(label) : label }${ msg ? `: ${msg}` : ''}`)
}
```

... but it's still nice how VM auto-converts, also for some reason their obj is much more detailed

TM:

![image](https://github.com/user-attachments/assets/7707bb1b-7b79-4e6b-b924-c9cda10f8da7)

VM:

![image](https://github.com/user-attachments/assets/b6c362bd-56c8-47e8-b191-60994e3483e7)

**Comment by derjanb** [comment]
Will be fixed at the next BETA version. Thanks for reporting.

Out of curiosity, the following test script is always logging `[object Object]` here. With no browser or script manager I get the stringified object logged.

![image](https://github.com/user-attachments/assets/72800fad-608f-461e-a0a3-f219b560d6da)

```
// ==UserScript==
// @name         234234.example.com
// @namespace    http://tampermonkey.net/
// @version      2024-08-29
// @description  try to take over the world!
// @author       You
// @match        https://*/*
// @grant        GM.xmlHttpRequest
// @connect      example.com
// ==/UserScript==

function consoleErr(label, msg) { console.error(`aa bb » ${label}${ msg ? `: ${msg}` : '' }`)}

GM.xmlHttpRequest({
    method: "GET",
    url: "https://234234.example.com",
    onerror: function(e) {
      consoleErr(e);
    }
});
```

**Comment by adamlui** [comment]
What's no browser or script manager mean? Also the code in FF TM shows the original TypeError to me

**Comment by adamlui** [comment]
> You said "in Violentmonkey it does not throw + properly logs the err obj". What does "properly" mean? I only get the string `[object Object]` being logged.

I get the pic I pasted 
![358860236-b6c362bd-56c8-47e8-b191-60994e3483e7](https://github.com/user-attachments/assets/e1a83f8f-15cf-44da-a727-5b0dbda22a9c)

Also my code is:

```js
                onerror: err => { log.err(err)
                    if (!config.proxyAPIenabled) appAlert(!config.openAIkey ? 'login' : ['openAInotWorking', 'suggestProxy'])
                    else if (get.reply.status != 'done') api.tryNew(get.reply)
                }
```

...  and I notice nothing happens after the error is logged, the next lines don't run as expected

![image](https://github.com/user-attachments/assets/e279c4e4-5768-47a4-bde8-6d9a3bc3a73e)

**Comment by adamlui** [comment]
I figured it out, the stream is considered done by the reader on error so i removed the condition

**Comment by derjanb** [comment]
Should be fixed at 5.3.6209 ([crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx)|xpi in review)

Please download the crx file linked above and drag and drop it to the extensions page `chrome://extensions` (after you've enabled 'Developer Mode').

For a quick fix please export your settings and scripts as zip or (JSON) file at the "Utilities" tab and import it back at the fixed BETA version.
